src/events.c

```diff
@@ -94,18 +131,13 @@ static void handle_e_bgkill_off_signal()
         send_bgkill_off_signal();
 }
 
-static void handle_e_input_device_attached(void)
-{      
-        inform_keyboard_attached(TRUE);
-}
- 
-static void handle_e_input_device_detached(void)
+static void handle_e_input_device_changed(void)
 {
-        if (get_input_device_state() != INPUT_DEVICE_ATTACHED) {
-               inform_keyboard_attached(FALSE);
-        }
+        reread_input_device_state();
+        inform_keyboard_attached(get_input_device_state() == INPUT_DEVICE_ATTACHED);
 }
 
+
```

Comment:
Is this extra line here on purpose or by chance?

Reply:
Removed in new push